src/features/Folding.ts

```diff
@@ -421,9 +421,9 @@ export class FoldingProvider implements vscode.FoldingRangeProvider {
     ): ITokenList {
         const result = [];
 
-        const emptyLine = /^[\s]+$/;
-        const startRegionText = /^#\s*region\b/;
-        const endRegionText = /^#\s*endregion\b/;
+        const emptyLine = /^\s*$/;
+        const startRegionText = /^#\s*region\b/i;
```

Comment:
I still think we shouldn't match `#    region` for this; I can imagine ordinary comments matching this, and I don't think I've ever seen any usage of a pragma that permitted whitespace like this.

Reply:
May want to hit up the VSCode team on that, as they're ALL like that

Reply:
I can only find examples where the whitespace comes before the `#`, like:
* [The extension folding snippets](https://github.com/Microsoft/vscode/search?q=%22%2F%23region%22&unscoped_q=%22%2F%23region%22)
* [The asks in the original issue](https://github.com/Microsoft/vscode/issues/12146)
* [The feature announcement](https://code.visualstudio.com/updates/v1_17#_folding-regions)
* [VSCode's folding tests](https://github.com/Microsoft/vscode/blob/0faa5d117cced88ab9d507cadf697c2acd3dcbff/src/vs/editor/contrib/folding/test/foldingModel.test.ts)
* [VSCode's test definition of `FoldingMarkers`](https://github.com/Microsoft/vscode/blob/059ead3c7bf1173477a73011d722b27384989448/src/vs/editor/contrib/folding/test/indentRangeProvider.test.ts#L150-L153)

Having an indented `#region` like:
```
if ($x)
{
    #region

       DoStuff

    #endregion
}
```

seems pretty reasonable to me.

But I feel like setting folds for
```
#   Region
if ($x)
{
    ... Blah
}

# EndRegion
```
could cause problems. Any line comment beginning with the word "region" would become a fold, meaning we would then look for `#   endregion`, and may not find one.

Reply:
Oh I am so sorry...I need glasses 👎 

-100 points to me.

Reply:
Haha, no I thought that's what was going on, just wanted to be sure 😄 

Thanks so much for this @glennsarti!